### PR TITLE
NOISSUE - Prevent infinite loop in lora-adapter if Redis init fail

### PR DESCRIPTION
**Pull request by manuio**
Signed-off-by: Manuel Imperiale <manuel.imperiale@gmail.com>

### CODE REVIEW

**Review by dborovcanin — DISMISSED** [review]

File: cmd/lora/main.go [hunk 1]
```diff
@@ -203,7 +203,9 @@ func subscribeToLoRaBroker(svc lora.Service, mc mqtt.Client, logger logger.Logge
 func subscribeToThingsES(svc lora.Service, client *r.Client, consumer string, logger logger.Logger) {
 	eventStore := redis.NewEventStore(svc, client, consumer, logger)
 	logger.Info("Subscribed to Redis Event Store")
-	eventStore.Subscribe("mainflux.things")
+	if err := eventStore.Subscribe("mainflux.things"); err != nil {
+		logger.Warn(fmt.Sprintf("Botstrap service failed to subscribe to event sourcing: %s", err))
```
Comment (anovakovic01) [inline cmd/lora/main.go]: Not Botstrap service, but LoRa adapter.

File: lora/redis/streams.go [hunk 2]
```diff
@@ -68,8 +70,11 @@ func NewEventStore(svc lora.Service, client *redis.Client, consumer string, log
 	}
 }
 
-func (es eventStore) Subscribe(subject string) {
-	es.client.XGroupCreateMkStream(stream, group, "$").Err()
+func (es eventStore) Subscribe(subject string) error {
+	if err := es.client.XGroupCreateMkStream(stream, group, "$").Err(); err != nil && err.Error() != exists {
+		return err
+	}
```
Comment (anovakovic01) [inline lora/redis/streams.go:77]: You shouldn't return here. If consumer group already exists, just continue.

**Review by anovakovic01 — CHANGES_REQUESTED** [review]

File: cmd/lora/main.go (hunk 1 above)
Comment (manuio) [reply cmd/lora/main.go]: done

File: lora/redis/streams.go (hunk 2 above)
Comment (manuio) [reply lora/redis/streams.go:77]: @dusanb94 what we do here? I thought that this is the fix

Comment (dborovcanin) [reply lora/redis/streams.go:77]: You return only if there is an error other than `BUSYGROUP Consumer Group name already exists`. If there is no error or error is exactly this one, you will proceed with your subscription.

Comment (manuio) [reply lora/redis/streams.go:77]: Right, this is correct @anovakovic01

Comment (anovakovic01) [reply lora/redis/streams.go:77]: Didn't see that part. IMHO, this line is too long. Can you break it into multiple lines? Also, is there predefined error for this in redis lib?

Comment (manuio) [reply lora/redis/streams.go:77]: I could not find any meaningful error. Line length fixed.

**Review by dborovcanin — APPROVED** [review]

**Review by anovakovic01 — APPROVED** [review]
LGTM!

**Pull request merged**